letsencrypt/client/revoker.py

```diff
@@ -6,146 +14,513 @@
+
+        """
+        success_list = []
+        try:
+            for cert in certs:
+                if self.no_confirm or revocation.confirm_revocation(cert):
+                    try:
+                        self._acme_revoke(cert)
+
+                        success_list.append(cert)
+                        revocation.success_revocation(cert)
+                    except errors.LetsEncryptClientError:
+                        # TODO: Improve error handling when networking is set...
+                        logging.error(
+                            "Unable to revoke cert:%s%s", os.linesep, str(cert))
+        finally:
+            if success_list:
+                self._remove_certs_keys(success_list)
+
+        return success_list
 
-    def remove_cert_key(self, cert):  # pylint: disable=no-self-use
+    def _acme_revoke(self, cert):
+        """Revoke the certificate with the ACME server.
+
+        :param cert: certificate to revoke
+        :type cert: :class:`letsencrypt.client.revoker.Cert`
+
+        :returns: TODO
+
+        """
+        try:
+            certificate = acme_util.ComparableX509(cert.cert)
+            with open(cert.backup_key_path, "rU") as backup_key_file:
+                key = Crypto.PublicKey.RSA.importKey(backup_key_file.read())
+
+        # If the key file doesn't exist... or is corrupted
+        except (OSError, IOError):
+            raise errors.LetsEncryptRevokerError("Unable to read key file")
+
+        # TODO: Catch error associated with already revoked and proceed.
+        return self.network.send_and_receive_expected(
+            messages.RevocationRequest.create(
+                certificate=certificate, key=key),
+            messages.Revocation)
+
+    def _remove_certs_keys(self, cert_list):  # pylint: disable=no-self-use
         """Remove certificate and key.
 
-        :param dict cert: Cert dict used throughout revocation
+        :param list cert_list: Must contain certs, each is of type
+            :class:`letsencrypt.client.revoker.Cert`
 
         """
-        list_file = os.path.join(self.config.cert_key_backup, "LIST")
-        list_file2 = os.path.join(self.config.cert_key_backup, "LIST.tmp")
+        # This must occur first, LIST is the official key
+        self._remove_certs_from_list(cert_list)
 
-        with open(list_file, 'rb') as orgfile:
-            csvreader = csv.reader(orgfile)
+        # Remove files
+        for cert in cert_list:
+            os.remove(cert.backup_path)
+            os.remove(cert.backup_key_path)
+
+    def _remove_certs_from_list(self, cert_list):  # pylint: disable=no-self-use
+        """Remove a certificate from the LIST file.
+
+        :param list cert_list: Must contain valid certs, each is of type
+            :class:`letsencrypt.client.revoker.Cert`
 
-            with open(list_file2, 'wb') as newfile:
+        """
+        list_path2 = os.path.join(self.config.cert_key_backup, "LIST.tmp")
```

Comment:
Thanks, one revision of this module I was planning on using info based on the file's existence, but I modified the design.